### ISSUE TITLE
Support 100 continue responses

### ISSUE THREAD

**Issue by matuszewskijan** [httprb/http#592]
It seems that HTTP isn't following the `100-continue` responses:
```
HTTP.headers('Content-Type' => 'application/json', 'Accept' => 'application/json').post('https://foo.bar/api', body: '{...}')
=> #<HTTP::Response/1.1 100 Continue {}>
```

when cURL does:
```
curl -v -X POST \
>   https://foo.bar/api \
>   -H 'Accept: application/json' \
>   -H 'Content-Type: application/json' \
>   -d '{...}'
...
> POST /api HTTP/1.1
> Host: foo.bar
> Accept: application/json
> Content-Type: application/json
> Content-Length: 294
> 
* upload completely sent off: 294 out of 294 bytes
< HTTP/1.1 100 Continue
< HTTP/1.1 200 OK
...
```

I would expect the HTTP response to be `<HTTP::Response/1.1 200 OK {}>`. 

Is it something what the HTTP gem should support?

**Comment by woutdegeyter** [comment]
Any feedback on this one, please? 
This is currently blocking us from building out an integration as the partner API is returning these 1xx codes.
@ixti 

Thanks already.

**Comment by ixti** [comment]
If I understand correctly, yes, that's something we should support. As far as I understand second response comes immediately after the 100 Continue - in other words 1 request, but 2 responses. Am I right?

**Comment by matuszewskijan** [comment]
Indeed, it seems that we are receiving 2 responses from one request. I'll see to prepare a PR to support this 100 continue code.

**Comment by matuszewskijan** [comment]
I've tried to add support for reading the second response and the only solution which I've found is a change in [`readpartial`](https://github.com/httprb/http/blob/master/lib/http/connection.rb#L86-L97) method:
```   
return unless @pending_response
   chunk = @parser.read(size)
   return chunk if chunk
   finished = (read_more(size) == :eof) || @parser.finished?
+
+  if finished && @buffer.include?("HTTP/1.1 100 Continue")
+    @parser.reset
+    finished = false
+  end
+
   chunk    = @parser.read(size)
   finish_response if finished
```

It seems to be very hacky for me, and the response `code` will be always `100`. 

I've spent few hours more to find anything better. Experimented with `@state.final_chunk?` in `parser.rb` but it seems to always return false for me(even when the response isn't in chunks at all).

Anyone is having any idea where we can put the logic to support this kind of scenario?

For more context - in `Net::Http` it's implemented in this way:  (ruby-2.5.1/lib/ruby/2.5.0/net/http.rb)
![image](https://user-images.githubusercontent.com/18028408/73978187-210b5180-492c-11ea-9b66-f4b06a9b75c2.png)
